src/test/suite/untitledFile.test.ts

```diff
@@ -0,0 +1,66 @@
+import * as vscode from "vscode";
+import { describe, beforeEach, it, afterEach } from "mocha";
+import { expect } from "chai";
+import { activate, sleep } from "./utils/helper";
+import { getLanguageFileExtension } from "../../runCompletion";
+import languages from "../../globals/languages";
+
+describe("Untitled file language detection", () => {
+  let editor: vscode.TextEditor | undefined;
+  let doc: vscode.TextDocument | undefined;
+
+  beforeEach(async () => {
+    const res = await activate();
+    editor = res?.editor;
+    doc = res?.doc;
+  });
+
+  afterEach(async () => {
+    await vscode.commands.executeCommand("workbench.action.closeActiveEditor");
+  });
+
+  const tableIt = async (input: string, fileExtension: string) => {
+    editor = editor as vscode.TextEditor;
+    doc = doc as vscode.TextDocument;
+    await editor.edit((edit) => {
+      edit.insert(new vscode.Position(0, 0), input);
+    });
+    await sleep(1000); // wait for editor to detect language
```

Comment:
isn't there any better way to confirm that the editor has detected the language? tests with `sleep`s are prone to being flaky.

Reply:
mmm ok then, lets hope it'll be stable enough 🙂